projects/ADuCM3029_demo_cn0428_cn0429/src/cli.c

```diff
@@ -0,0 +1,986 @@
+	UART_TX("                                  <time> = <50-32000>"_EOS);
+	UART_TX("                                  ADC performs avg of 10 samples at this interval"_EOS);
+	UART_TX("startmeas                      -Start ADC sampling at configured interval"_EOS);
+	UART_TX("stopmeas                       -Stop ADC sampling the sensor"_EOS);
+	UART_TX("setrtia <rtia>                 -Configure TIA resistance"_EOS);
+	UART_TX("                                  <rtia> = <0-26> equals 0R - 512k"_EOS);
+	UART_TX("                                  see user guide for detailed table"_EOS);
+	UART_TX("setrload <rload>               -Configure sensor load resistor"_EOS);
+	UART_TX("                                  <rload> = <0-7> equals 0R - 3k6"_EOS);
+	UART_TX("                                  see user guide for detailed table"_EOS);
+	UART_TX("setvbias <vbias>               -Configure sensor bias voltage"_EOS);
+	UART_TX("                                  <vbias> = bias voltage in mV"_EOS);
+	UART_TX("setsensitivity <sens>          -Configure sensitivity"_EOS);
+	UART_TX("                                  <sens> = sensitivity in nA/ppm"_EOS);
+	UART_TX("runeis                         -Run impedance spectroscopy"_EOS);
+	UART_TX("readeis                        -Read impedance spectroscopy results"_EOS);
+	UART_TX("readeisfull                    -Read impedance spectroscopy FULL results"_EOS);
+	UART_TX("readrcal                       -Read 200R calibration resistor"_EOS);
+	UART_TX("runpulse                       -Run pulse test. Set amplitude and duration first!"_EOS);
+	UART_TX("readpulse                      -Read pulse test results"_EOS);
+	UART_TX("pulseamplitude <ampl>          -Set amplitude for pulse test"_EOS);
+	UART_TX("                                  <ampl> = amplitude in mV, 1 - 3 mV"_EOS);
+	UART_TX("pulseduration <dur>            -Set duration for pulse test"_EOS);
+	UART_TX("                                  <dur> = duration in ms, 10 to 200"_EOS);
+	UART_TX("readsensors                    -Read the ppb value of all sensors"_EOS);
+	UART_TX("setupdaterate <time>           -Configure update rate for stream mode"_EOS);
+	UART_TX("stopread                       -Stop sensor data reading"_EOS);
+	UART_TX("\r\n"_EOS);
+}
+
+/**
+   @brief Finds the next command line argument
+
+   @param args - pointer to the arguments on the command line.
+
+   @return pointer to the next argument.
+
+**/
+uint8_t *FindArgv(uint8_t *args)
+{
+	uint8_t *p = args;
+	int     fl = 0;
+
+	while (*p != 0) {
+		if ((*p == _SPC))
+			fl = 1;
+		else if (fl)
+			break;
+		p++;
+	}
+
+	return p;
+}
+
+/**
+   @brief Separates a command line argument
+
+    @param dst - pointer to a buffer where the argument will be copied
+    @param args - pointer to the current position of the command line .
+
+   @return none
+
+**/
+void GetArgv(char *dst, uint8_t *args)
+{
+	uint8_t  *s = args;
+	char     *d = dst;
+
+	while (*s) {
```

Comment:
similar to comment https://github.com/analogdevicesinc/EVAL-ADICUP3029/pull/12/commits/4f5694854b122faead93c95e10cae42d6207dad4#r243189944  this byte-to-byte lookup is un-needed;
you can use argv and argc

Reply:
See previous comment https://github.com/analogdevicesinc/EVAL-ADICUP3029/pull/12#discussion_r243378101